core/templates/dev/head/player/PlayerServices.js

```diff
@@ -163,16 +163,28 @@ oppia.factory('oppiaPlayerService', [
 
   var _updateStatus = function(newParams, newStateName) {
     // TODO(sll): Do this more incrementally.
-    learnerParamsService.init(newParams);
-    _currentStateName = newStateName;
-    stateHistory.push(_currentStateName);
+    if (newParams !== null) {
```

Comment:
I've refactored this -- it was originally just defense in depth to prevent the params/stateName from being overwritten erroneously. I moved the check to applyCachedParamUpdates() instead; PTAL.